### PR TITLE
[WIP] python: add alpha release of 3.6

### PR DESCRIPTION
**Pull request by kragniz**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useChroot](http://nixos.org/nixos/manual/options.html#opt-nix.useChroot) on NixOS,
    or option `build-use-chroot` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [x] NixOS
  - [ ] OS X
  - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

This might be jumping the gun a little, but I needed to use 3.6 locally
and thought I may as well push it upstream.

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -5842,6 +5842,10 @@ in
     inherit (darwin) CF configd;
     self = python35;
   });
+  python36 = hiPrio (callPackage ../development/interpreters/python/3.6 {
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: this shouldn't be `hiPrio`. Python 3.5 should remain more prominent.

Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: actually, 3.4 shouldn't be `hiPrio` either but that is out of scope.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -9885,6 +9889,11 @@ in
     self = python35Packages;
   });
 
+  python36Packages = recurseIntoAttrs (callPackage ./python-packages.nix {
```
Comment (FRidh) [inline pkgs/top-level/all-packages.nix]: also not `recurseIntoAttrs` here. I don't think we should make the packages visible yet.

**Comment by FRidh** [comment]
Personally I don't mind having it in already, although typically we only keep stable releases.
Note that there is also some work to be done in `python-packages.nix`, E.g. `isPy36`.

**Comment by kragniz** [comment]
Thanks for the review, I've updated the patch

**Comment by FRidh** [comment]
> Note that there is also some work to be done in python-packages.nix, E.g. isPy36.

**Comment by FRidh** [comment]
pushed as 7c75fa4bc20492ac52c0b978a777d55e5844c9a2
